### ISSUE TITLE
`swarm logs` shouldn't require instance id in all cases

### ISSUE THREAD

**Issue by teemow** [giantswarm/cli#57]
We require users to give an instance ID when calling `swarm logs`.

If a component runs only on one instance, We shouldn't require this. Instead we could allow the user to simply give the <app/service/component> name as an argument. This way the user doesn't have to find out himself which instance the component is running on.

This would require us recognize whether the given argument is an instance ID or an <app/service/component> string, which should be easy.

We would need to handle the case where the user gives an <app/service/component> string, but there is more than one instance running. An easy way would be this:

```
$ swarm logs myapp/myservice/mycomponent
There is more than one instance running this component. Please use the following syntax:

    swarm logs <instance_id>

You can list all instances using

    swarm status myapp
```

A more advanced and more convenient way would be:

```
$ swarm logs myapp/myservice/mycomponent
There are 3 instances running this component:

9c79161b-baec-447b-9a8e-230206268570
b4405a86-958e-4a46-ac14-41404c5e17bd
c4a38g02-472d-4c38-a5b1-1b3a69cdc3d2

Please use the following syntax to access logs:

    swarm logs <instance_id>
```

Original Author: @marian
Created at 09-11-2014 13:02:57
Gitlab: [giantswarm/cli!97](https://git.giantswarm.io/giantswarm/cli/issues/97)

**Comment by marians** [comment]
This only really makes sense if it also works in the case that a component has more than one instance. So it is depending on log aggregation. See #45

**Comment by ZeissS** [comment]
Closing then, as it will be fixed by #45, if implemented.
